lib/ansible/modules/source_control/git.py

```diff
@@ -432,8 +447,28 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
                         "HEAD, branches, tags or in combination with refspec.")
     if reference:
         cmd.extend(['--reference', str(reference)])
+    separate_git_dir_fallback = False
+    if separate_git_dir is not None and bare:
+        module.warn("Ignoring separate_git_dir argument. "
+                    "Can not do bare clone with argument separate_git_dir.")
+    elif separate_git_dir:
+        if os.path.exists(os.path.abspath(separate_git_dir)):
+            module.fail_json(msg='Separate-git-dir path %s already exists.' % os.path.abspath(separate_git_dir))
```

Comment:
You calculate `os.path.abspath(separate_git_dir)` in more than two places. Save it to a variable and reuse, please.